src/qforte/utils/symmetry_analysis.py

```diff
@@ -0,0 +1,170 @@
+import qforte as qf
+import numpy as np
+
+def symmetry_check(n_qubits, Uprep, pool, tops, tamps, compact, qubit, trotter_number, irreps, orb_irreps_to_int):
+    """
+    This function performs a symmetry analysis of the provided quantum state.
+    In particular, it computes the expectation values and variances of the
+    particle number N, projection of total spin on z axis Sz, and total spin
+    squared S^2 operators. In addition, it computes the weight contribution
+    to the wavefunction of the various N, Sz, and spatial symmetry sectors.
+
+    Arguments
+    =========
+
+    n_qubits: int
+              The number of qubits of the system.
+
+    Uprep: Circuit
+           The circuit that transforms the |0..0> state to the reference one.
+
+    pool: SQOpPool
+        Pool of second-quantized operators used in constructing the ansatz.
+
+    tops: list of int
+        List of indices denoting the operators from the pool that have been selected for the ansatz.
+
+    tamps: list of float
+        Amplitudes corresponding to the second-quantized operators in tops.
+
+    compact: boolean
+        Controls the use of standard vs compact excitation circuits.
+
+    qubit: boolean
+        Controls the use of fermionic vs qubit anti-Hermitian excitations.
+
+    trotter_number: int
+        The number of Trotter steps used to approximate the ansatz. Currently implemented only for compact = False.
+
+    irreps: list of str
+        List that holds the irreps of the group in the Cotton ordering.
+
+    orb_irreps_to_int: list of int
+        Integer n indexes the irrep of spatial orbital n.
+    """
+
+    # Generate the quantum state defined by the Uprep, tops, tamps, compact, qubit and trotter_number
+    qc = qf.Computer(n_qubits)
+    qc.apply_circuit(Uprep)
+    if compact:
```

Comment:
I _really_ don't like that this code needs to know the details of how to construct the computer. Can we have the code take in the Computer as an argument, instead?

Reply:
I do agree with you that the code looks ugly. If we pass the Computer as an argument, whenever we want to use this function we will need to do the following:
```
U = self.ansatz_circuit()
comp = qf.Computer(self._nqb)
comp.apply_circuit(self._Uprep)
comp.apply_circuit(U)
```
Indeed, much more concise. Should I proceed with the simplifications?

Reply:
You can make it even shorter.
```
comp = qf.Computer(self._nqb)
comp.apply_circuit(self.build_Uvqc())
```

Yes, simplify ahead.